### PR TITLE
Various PHP related fixes and updates

### PR DESCRIPTION
**Pull request by kornrunner**
`flatten` is unused, but had an error (reference to non-existent foo).

`array_concat` didn't account for arrays to concatenate/merge

### CODE REVIEW

**Comment by kornrunner** [comment]
Pushed a transpiler fixup as well for `.length` (wrongly transpiled to php).

Perhaps we could add a simple syntax check to travis, something like:
```sh
$ find php/ -type f -iname \*.php -exec php -l {} \;
```

**Comment by kornrunner** [comment]
This includes fixes for #924 and #923 (and some not yet reported)

**Comment by hanoncs** [comment]
Thanks. Just got another syntax error in the kucoin.php file.

File: js/bter.js [hunk 1]
```diff
@@ -63,7 +63,12 @@ module.exports = class bter extends Exchange {
 
     async fetchMarkets () {
         let response = await this.publicGetMarketinfo ();
-        let markets = response['pairs'];
+        let markets = undefined;
+        if ('pairs' in response) {
+            markets = response['pairs'];
+        } else {
+            markets = [];
```
Comment (kroitor) [inline js/bter.js]: I guess, I'm going to add / throw an exception here, what do you think?

Comment (kornrunner) [reply js/bter.js]: Makes sense

**Pull request merged**